src/ClusterBootstrap/cloud_init_aztools.py

```diff
@@ -267,12 +270,10 @@ def gen_machine_list_4_deploy_action(complementary_file_name, config):
                     continue
                 cc["machines"][vmname][k] = v
             if "kube_label_groups" not in spec:
-                if "infra" in spec["role"]:
-                    cc["machines"][vmname]["kube_label_groups"] = [
-                        "all", "etcd_node", "etcd_node_1"]
-                elif "worker" in spec["role"]:
-                    cc["machines"][vmname]["kube_label_groups"] = [
-                        "all", "worker_node"]
+                cc["machines"][vmname]["kube_label_groups"] = []
+                for role in spec["role"]:
+                    if role in config["default_kube_labels_by_node_role"]:
+                        cc["machines"][vmname]["kube_label_groups"] += role,
```

Comment:
can we do `cc["machines"][vmname]["kube_label_groups"].append(role)` for clarity?